pkg/providers/openstack/delete_image.go

```diff
@@ -2,8 +2,56 @@ package openstack
 
 import (
 	"github.com/emc-advanced-dev/pkg/errors"
+	"github.com/emc-advanced-dev/unik/pkg/types"
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack/imageservice/v2/images"
 )
 
 func (p *OpenstackProvider) DeleteImage(id string, force bool) error {
-	return errors.New("not yet supportded for openstack", nil)
+	image, err := p.GetImage(id)
+	if err != nil {
+		return errors.New("retrieving image", err)
+	}
+
+	// Delete instances of this image.
+	instances, err := p.ListInstances()
+	if err != nil {
+		return errors.New("retrieving list of instances", err)
+	}
+	for _, instance := range instances {
+		if instance.ImageId == image.Id {
+			if !force {
+				return errors.New("instance "+instance.Id+" found which uses image "+image.Id+"; try again with force=true", nil)
+			} else {
+				err = p.DeleteInstance(instance.Id, true)
+				if err != nil {
+					return errors.New("failed to delete instance "+instance.Id+" which is using image "+image.Id, err)
+				}
+			}
+		}
+	}
+
+	clientGlance, err := p.newClientGlance()
+	if err != nil {
+		return err
+	}
+
+	if err := deleteImage(clientGlance, image.Id); err != nil {
+		return errors.New("failed deleting image "+image.Id, err)
+	}
+
+	// Update state.
+	if err := p.state.ModifyImages(func(images map[string]*types.Image) error {
```

Comment:
At first it was a bit confusing because you have `image` var here and referring to `images` package in l55. Do you think you could change it here to `imageList`?